roboteam_ai/src/coach/defence/DefencePositionCoach.cpp

```diff
@@ -322,6 +327,65 @@ Vector2 DefencePositionCoach::findPositionForBlockBall(const Line &blockLine) {
     }
     return getPosOnLine(blockLine, 0.3);
 }
+
+world::WorldData DefencePositionCoach::getTheirAttackers(const world::WorldData &world) {
+    std::vector<world::Robot> theirAttackers;
+    for (const world::Robot &robot :world.them) {
+        // we remove any attackers that are outside of the field or in our defence area
+        if (world::field->pointIsInDefenceArea(robot.pos, true, 0.04)) {
+            continue;
```

Comment:
continues can give undefined behaviour. it is better not to use them. You could better take the two boolean statements and push attackers into your vector when they are both false.  (also saves 4 lines)

Reply:
fixed it. Think I was tired here, this seems stupid :P